cmd/virt-launcher/virt-launcher.go

```diff
@@ -496,6 +513,12 @@ func main() {
 			finalShutdownCallback,
 			gracefulShutdownCallback)
 
+		ctx, cancelCtx := context.WithCancel(context.Background())
+		defer cancelCtx()
+
+		if err := runDownwardMetricsVirtioServer(ctx); err != nil {
+			log.Log.Reason(err).Error("failed to start the DownwardMetrics server")
```

Comment:
We are not exiting here?
Is it not a requirement?

Reply:
Sorry for the silly question, but could you please elaborate what "existing" mean? 

Does it mean to panic? For instance:

```go
if err := runDownwardMetricsVirtioServer(ctx); err != nil {
    panic(fmt.Errorf("failed to start the DownwardMetrics server %v", err))
}
```

Or do you mean something like closing the `signalStopChan` so the launcher gets exited?. For instance:

```go
if err := runDownwardMetricsVirtioServer(ctx); err != nil {
    log.Log.Reason(err).Error("failed to start the DownwardMetrics server")
    close(signalStopChan)
}	
```

Reply:
Took second option. Please, confirm this is what you have asked for.

Reply:
And yes, let's state this is a requirement, if the server fails to start, let's exit.

Reply:
I don't think there's a point in continuing to start the VM with a closed `signalStopChan`. You should better `panic` instead. Can't you start the server earlier, i.e. around line 482 before starting to listen for signals?

Reply:
You could create the new context for rate limiting closer to where it's used though. There is no need to create it at this level anymore.

Reply:
Done.